codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerBuilderGeneratorWithoutPublicConstrainedTypes.kt

```diff
@@ -160,6 +164,10 @@ class ServerBuilderGeneratorWithoutPublicConstrainedTypes(
                 withBlock("$memberName: self.$memberName", ",") {
                     serverBuilderConstraintViolations.forMember(member)?.also {
                         rust(".ok_or(ConstraintViolation::${it.name()})?")
+                    } ?: run {
```

Comment:
Same here, we need to inject the writable before constraint checking.

We also need unit tests that exercise this builder type.